common/NMEA_parser.h

```
@@ -0,0 +1,131 @@
+            if (!words.empty() && words.at(0) == NMEA_sentence_id) {          
+                break;
+            } else {
+                words.clear();
+            }       
+        }
+        return words;
+    }
+
+    
+    public:
+
+    static bool parseTextGPS(const std::string   &toParse,            // text to parse
+                             degree_t            &latitude,           // latitude
+                             arcminute_t         &latitudeMinutes,    // minute part of latitude
+                             std::string         &latDirection,       // latitude direction North or South (N | S)
+                             degree_t            &longitude,          // longitude
+                             arcminute_t         &longitudeMinutes,   // minute part of longitude
+                             std::string         &longDirection,      // longitude direction East or West (E | W)
+                             meter_t             &altitude,           // altitude in meters
+                             meters_per_second_t &velocity
+                             ) {
+
+        using namespace std;
+        try {
+            vector<string> elements= parseNMEAstring(toParse, "GNGGA"); // parse string 
+            latitude = degree_t(std::stod(elements[2].substr(0,2)));
+            latitudeMinutes = arcminute_t(std::stod(elements[2].substr(2,8)));     
+            latDirection = elements[3];
+            longitude = degree_t(std::stod(elements[4].substr(0,3)));
+            longitudeMinutes = arcminute_t(std::stod(elements[4].substr(3,9)));
+            longDirection = elements[5];
+            altitude = meter_t(stod(elements[9]));
+
+            // prse GNRMC for velocity
+            vector<string> elementsRMC= parseNMEAstring(toParse, "GNRMC"); // parse string 
+            velocity = knot_t(stod(elementsRMC[7])); 
+        } catch(...) {
+            return false;
+        }   
+        return true;
```

I think @neworderofjamie has said something similar elsewhere, but I would just let the exceptions be thrown rather than returning a boolean. It's less code and means you don't have to rely on the user remembering to check the return value.